src/resolve-info.ts

```diff
@@ -61,51 +71,85 @@ export function isLeafField(obj: LeafField | FieldExpansion): obj is LeafField {
  * that returns the computed resolveInfo. This thunk is registered in
  * context.dependencies for the field's resolveInfoName
  */
-export function createResolveInfoThunk({
-  schema,
-  fragments,
-  operation,
-  parentType,
-  fieldName,
-  fieldType,
-  fieldNodes
-}: {
-  schema: GraphQLResolveInfo["schema"];
-  fragments: GraphQLResolveInfo["fragments"];
-  operation: GraphQLResolveInfo["operation"];
-  parentType: GraphQLObjectType;
-  fieldType: GraphQLOutputType;
-  fieldName: string;
-  fieldNodes: FieldNode[];
-}) {
-  const fieldExpansion: FieldExpansion | LeafField = {};
-
-  for (const fieldNode of fieldNodes) {
-    deepMerge(
-      fieldExpansion,
-      memoizedExpandFieldNode(schema, fragments, fieldNode, fieldType)
-    );
+export function createResolveInfoThunk(
+  {
+    schema,
+    fragments,
+    operation,
+    parentType,
+    fieldName,
+    fieldType,
+    fieldNodes
+  }: {
+    schema: GraphQLResolveInfo["schema"];
+    fragments: GraphQLResolveInfo["fragments"];
+    operation: GraphQLResolveInfo["operation"];
+    parentType: GraphQLObjectType;
+    fieldType: GraphQLOutputType;
+    fieldName: string;
+    fieldNodes: FieldNode[];
+  },
+  enricher?: (i: ResolveInfoEnricherInput) => any
```

Comment:
nit: returns `object | undefined` instead of `any`. And as we are spreading it below, `undefined` is actually fine.